File: 876/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        
        ListNode node1 = head;
        ListNode node2 = head;
        
        while(node1 != null && node1.next != null) {
            node1 = node1.next.next;
            node2 = node2.next;
        }
        
        
        return node2;
    }
}

File: 876/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp1 = head
        temp2 = head 
        
        while temp1 and temp1.next:
          
            temp1 = temp1.next.next
            temp2 = temp2.next
        return temp2